mesheryctl/pkg/utils/healthcheck.go

```diff
@@ -290,3 +290,32 @@ func WaitForPodRunning(c *meshkitkube.Client, desiredPod, namespace string, time
 
 	return pollForPodRunning(c, namespace, desiredPodName, time.Duration(timeout)*time.Second)
 }
+
+// Returns condition function to indicate that the `namespace` does not exist anymore.
+func isNamespaceDeleted(c *meshkitkube.Client, namespace string) wait.ConditionFunc {
+	return func() (bool, error) {
+		namespaces, err := c.KubeClient.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		// Check if namespace exists in namespaces list
+		for _, ns := range namespaces.Items {
+			if ns.Name == namespace {
+				return false, nil
+			}
+		}
+
+		return true, nil
+	}
+}
+
+// Poll up to timeout seconds every 5 seconds until the namespace no more exists.
+func pollForNamespaceDeleted(c *meshkitkube.Client, namespace string, timeout time.Duration) error {
+	return wait.PollImmediate(5*time.Second, timeout, isNamespaceDeleted(c, namespace))
```

Comment:
We don't want to PollImmediate here, we should do it interval-based, we discussed right? Deleting namespace takes a good time, we can take a lease here and do it at intervals.

Reply:
This does poll every 5 seconds, it only starts with a poll. I can replace it with `Poll` which starts with a wait which makes more sense because it's not very useful to poll in the start since it'll definitely take more time to stop :+1: